plugin/storage/grpc/factory.go

```diff
@@ -205,8 +212,22 @@ func (f *Factory) CreateArchiveSpanWriter() (spanstore.Writer, error) {
 // Close closes the resources held by the factory
 func (f *Factory) Close() error {
 	var errs []error
-	if f.remoteConn != nil {
-		errs = append(errs, f.remoteConn.Close())
+	if f.tracedRemoteConn != nil {
+		errs = append(errs, f.tracedRemoteConn.Close())
+	}
+	if f.untracedRemoteConn != nil {
+		errs = append(errs, f.untracedRemoteConn.Close())
 	}
 	return errors.Join(errs...)
 }
+
+func getTelset(logger *zap.Logger, tracerProvider trace.TracerProvider) component.TelemetrySettings {
+	return component.TelemetrySettings{
+		Logger:         logger,
+		TracerProvider: tracerProvider,
+		// TODO needs to be joined with the metricsFactory
+		LeveledMeterProvider: func(_ configtelemetry.Level) metric.MeterProvider {
+			return noopmetric.NewMeterProvider()
```

Comment:
As you can see below we're passing noop provider, that's not good. But we also cannot make up a new provider here, it should be passed down from OTEL framework, and it makes sense to pass it down in our telset struct.

Reply:
Sounds good! Added https://github.com/jaegertracing/jaeger/issues/6127